mindexer/utils/query.py

```diff
@@ -161,6 +161,25 @@ def index_intersect(self, index):
 
         return query
 
+    def index_number_key_qurey(self, index):
+        """
+            return the query that can be used to determine the number of 
+            index key needs to be examined.
+        """
+        query = Query()
+
+        for field in index:
+            if field in self.filter:
+                if isinstance(self.filter[field], dict):
+                    query.add_predicate({field: self.filter[field]})
+                    break
+                else:
+                    query.add_predicate({field: self.filter[field]})
+            else:
+                break
+
+        return query
+
```

Comment:
Can you add a few unit tests under `tests/test_query.py` to confirm this function does what it is supposed to?